zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -230,16 +235,34 @@ public void sync(AuthenticationInfo subject) throws IOException {
   }
 
   private void pushNotes(AuthenticationInfo subject, List<String> ids, NotebookRepo localRepo,
-      NotebookRepo remoteRepo) {
+      NotebookRepo remoteRepo, boolean setPermissions) {
     for (String id : ids) {
       try {
         remoteRepo.save(localRepo.get(id, subject), subject);
+        if (setPermissions) {
+          makePrivate(id, subject);
+        }
       } catch (IOException e) {
         LOG.error("Failed to push note to storage, moving onto next one", e);
       }
     }
   }
 
+  private void makePrivate(String noteId, AuthenticationInfo subject) {
+    if (subject != null && !"anonymous".equals(subject.getUser())) {
```

Comment:
makes sense, can be reused to make `anonymous` consistent in usage over codebase. I'll change now